app/services/base_type_service.rb

```diff
@@ -75,13 +75,17 @@ def set_params_and_validate(params)
     # (Regression #28400)
     set_attribute_groups(params) if params[:attribute_groups]
 
-    set_subject_pattern(params) if params[:subject_configuration]
-
     # This should go before `set_scalar_params` call to get the
     # project_ids, custom_field_ids diffs from the type and the params.
     # For determining the active custom fields for the type, it is necessary
     # to know whether the type is a milestone or not.
     set_milestone_param(params) unless params[:is_milestone].nil?
+
+    # Set patterns includes a data validation before assigning the value to the attribute.
+    # A failure should return a failure.
+    set_patterns(params) unless params[:patterns].nil?
```

Comment:
:green_circle: I'd prefer to avoid negations. Therefore I'd suggest either of

```suggestion
    set_patterns(params) if params[:patterns]
```

or

```suggestion
    set_patterns(params) if params[:patterns].present?
```

None of the two is exactly equal to your code w.r.t. semantics, thought I'd assume that we either expect a hash with some keys or `nil`, in which case the behaviour should be the same.

Reply:
yep, agreed